substrate/frame/people/src/lib.rs

```diff
@@ -0,0 +1,2007 @@
+				head = head.checked_add(1).unwrap_or(0);
+				let second_key_page = OnboardingQueue::<T>::take(head);
+				defensive_assert!(!second_key_page.is_empty());
+				keys_to_include.extend(second_key_page.into_iter());
+			}
+
+			let onboarding_size = OnboardingSize::<T>::get();
+
+			let (to_include, ring_filled) = Self::should_onboard_people(
+				top_ring_index,
+				&ring_status,
+				open_slots,
+				keys_to_include.len().saturated_into(),
+				onboarding_size,
+			)
+			.ok_or(Error::<T>::Incomplete)?;
+
+			let mut remaining_keys = keys_to_include.split_off(to_include as usize);
+			for key in keys_to_include.into_iter() {
+				let personal_id = Keys::<T>::get(&key).defensive().ok_or(Error::<T>::NotPerson)?;
+				let mut record =
+					People::<T>::get(personal_id).defensive().ok_or(Error::<T>::KeyNotFound)?;
+				record.position = RingPosition::Included {
+					ring_index: top_ring_index,
+					ring_position: keys.len().saturated_into(),
+					scheduled_for_removal: false,
+				};
+				People::<T>::insert(personal_id, record);
+				keys.try_push(key).map_err(|_| Error::<T>::TooManyMembers)?;
+			}
+			RingKeys::<T>::insert(top_ring_index, keys);
+			ActiveMembers::<T>::mutate(|active| *active = active.saturating_add(to_include));
+			ring_status.total = ring_status.total.saturating_add(to_include);
+			RingKeysStatus::<T>::insert(top_ring_index, ring_status);
+
+			// Update the top ring index if this onboarding round filled the current ring.
+			if ring_filled {
+				CurrentRingIndex::<T>::mutate(|i| i.saturating_inc());
+			}
+
+			if remaining_keys.len() > T::OnboardingQueuePageSize::get() as usize {
+				let split_idx =
+					remaining_keys.len().saturating_sub(T::OnboardingQueuePageSize::get() as usize);
+				let second_page_keys: BoundedVec<MemberOf<T>, T::OnboardingQueuePageSize> =
+					remaining_keys
+						.split_off(split_idx)
+						.try_into()
+						.expect("the list shrunk so it must fit; qed");
+				let remaining_keys: BoundedVec<MemberOf<T>, T::OnboardingQueuePageSize> =
+					remaining_keys.try_into().expect("the list shrunk so it must fit; qed");
+				OnboardingQueue::<T>::insert(old_head, remaining_keys);
+				OnboardingQueue::<T>::insert(head, second_page_keys);
+				QueuePageIndices::<T>::put((old_head, tail));
+			} else if !remaining_keys.is_empty() {
+				let remaining_keys: BoundedVec<MemberOf<T>, T::OnboardingQueuePageSize> =
+					remaining_keys.try_into().expect("the list shrunk so it must fit; qed");
+				OnboardingQueue::<T>::insert(head, remaining_keys);
+				QueuePageIndices::<T>::put((head, tail));
+			} else {
+				// We have nothing to put back into the queue, so if this isn't the last page, move
+				// the head to the next page of the queue.
+				if head != tail {
+					head = head.checked_add(1).unwrap_or(0);
+				}
+				QueuePageIndices::<T>::put((head, tail));
+			}
+
+			Ok(())
+		}
+
+		/// Task that merges two rings. In order for the rings to be eligible for merging, they must
```

Comment:
It's not used in any of the hooks, so I guess it's not a task actually.

Reply:
Rephrased.